lib/httpClient.js

```diff
@@ -173,10 +179,32 @@ Client.prototype.destroy = function () {
   if (!this.destroyed) {
     this.destroyed = true
     this.timeoutTicker.clear()
-    if (this.rate) clearInterval(this.rateInterval)
-    this.emit('done')
-    this._destroyConnection()
+
+    if (this.rate) {
+      clearInterval(this.rateInterval)
+    }
+
+    return new Promise((resolve) => {
+      if (!this.reqInProgress) {
+        this.emit('done')
+        this._destroyConnection()
+
+        return resolve()
+      }
+
+      this.on('response', () => {
+        this.emit('done')
+        this._destroyConnection()
+        this.removeAllListeners('response')
+
+        resolve()
+      })
+    })
   }
+
+  return new Promise((resolve) => {
+    resolve()
+  })
```

Comment:
This whole module is based on the concept of callbacks, not promises. Would you mind using callbacks instead?
Also the above code _might_ leak memory in case of an error.

Would you mind adding a few comment/description on why we are doing all of this?

Reply:
I want to know the exact number of requests sent